util/session/sessionmanager.go

```diff
@@ -502,9 +506,17 @@ func WithAuthMiddleware(disabled bool, authn TokenVerifier, next http.Handler) h
 				return
 			}
 			ctx := r.Context()
+
+			// Assert that claims is of type jwt.MapClaims
+			mapClaims, ok := claims.(jwt.MapClaims)
+			if !ok {
+				http.Error(w, "Invalid claims type", http.StatusUnauthorized)
+				return
+			}
```

Comment:
I don't think we should use MapClaims. See the comment on the RegisterClaims object
> // RegisteredClaims are a structured version of the JWT Claims Set,
// restricted to Registered Claim Names, as referenced at
// https://datatracker.ietf.org/doc/html/rfc7519#section-4.1
//
// This type can be used on its own, but then additional private and
// public claims embedded in the JWT will not be parsed. The typical usecase
// therefore is to embedded this in a user-defined claim type.

basically, define a new struct 
```go
type ArgoClaims struct {
	RegisteredClaims
	OtherClaim string `json:"othet,omitempty"`
	...
```